das/avail/avail.go

```diff
@@ -0,0 +1,235 @@
+package avail
+
+import (
+	"context"
+	"fmt"
+
+	"time"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
+	gsrpc_types "github.com/centrifuge/go-substrate-rpc-client/v4/types"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types/codec"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/das/dastree"
+	"github.com/vedhavyas/go-subkey"
+)
+
+// AvailMessageHeaderFlag indicates that this data is a Blob Pointer
+// which will be used to retrieve data from Avail
+const AvailMessageHeaderFlag byte = 0x0a
+
+func IsAvailMessageHeaderByte(header byte) bool {
+	return (AvailMessageHeaderFlag & header) > 0
+}
+
+type BridgdeApiResponse struct {
+	BlobRoot           gsrpc_types.Hash   `json:"blobRoot"`
+	BlockHash          gsrpc_types.Hash   `json:"blockHash"`
+	BridgeRoot         gsrpc_types.Hash   `json:"bridgeRoot"`
+	DataRoot           gsrpc_types.Hash   `json:"dataRoot"`
+	DataRootCommitment gsrpc_types.Hash   `json:"dataRootCommitment"`
+	DataRootIndex      uint64             `json:"dataRootIndex"`
+	DataRootProof      []gsrpc_types.Hash `json:"dataRootProof"`
+	Leaf               gsrpc_types.Hash   `json:"leaf"`
+	LeafIndex          uint64             `json:"leafIndex"`
+	LeafProof          []gsrpc_types.Hash `json:"leafProof"`
+	RangeHash          gsrpc_types.Hash   `json:"rangeHash"`
+}
```

Comment:
i have written a much cleaner version for this code in OP. Can you refer that and clean this file up a bit please.

Reply:
Will cover this in my next pr